### DISCUSSION TITLE
Run Waze native on Tesla Android with USB GPS receiver

### DISCUSSION THREAD

**Discussion by josbmw** [tesla-android/issue-tracker#271] [General]
Hi,

I know there have been questions about this already.
It looks like people just want to be able to run Waze native on Tesla Android without Carlinkit or Android Auto or Carplay.

For example I use Graphene OS on my phone (a privacy hardened OS).
This OS does not support Android Auto due to the many permissions Android Auto wants.

But I can install the Waze APK on Tesla Android.
When I enable USB tethering, I have internet on Tesla Android and Waze works.
But there is no GPS in the Pi and you can't share GPS data from Android.

Can I just hook up a USB GPS receiver to the Pi and use location data?

Jos

**Comment by mikegapinski** [comment]
Not officially and it does not work well. I've commited to delivering the GPS integration in the next major release ;)

**Reply by josbmw** [reply]
Thanks Mike,

I ordered a USB GPS receiver and installed the GPS Connector app on the Pi plus modified the developer setting required.
Will let you know how it goes when I have the GPS receiver.

Jos

**Reply by mikegapinski** [reply]
If you like wasting money on those crappy receivers. Makes no sense, especially given how close the official integration is

**Reply by josbmw** [reply]
Ordered from Amazon, can always return when not ok.
What GPS receiver do you plan to use if not USB?

**Reply by mikegapinski** [reply]
The car itself, the browser can get coordinates from the car every second, I am feeding them back to android and injecting them via a custom hardware gps driver. No mock locations etc

**Reply by josbmw** [reply]
Nice! Looking forward to the final solution.

**Reply by galosre** [reply]
I just achieved this with an old receiver I had, it is working, but Waze, so far get no sound and the update in terms of speed is not instant . So it is better to wait for Mike's solution, which is much more elegant .and functional.

**Reply by josbmw** [reply]
Got my GPS receiver today. Works perfect with USB tethering to my smartphone for the data and GPS Integration app for the receiver.
Of course native location integration from the Tesla browser is a nicer solution, but for the moment this is a good alternative.

Only one issue: when I test it with a Macbook everything works and is very stable.
But in the car, the Tesla browser often crashes (Model S Performance 2019). 
The Pi is sitting on the passenger seat for testing and keeps running without a problem.

I read about issues with the Wifi signal with thses older cars, so I tried placing the Pi in other locarions.
Does not really improve the situation, browser keeps crashing.

Any ideas?

**Reply by mikegapinski** [reply]
Try accessing the frontend via the ip address and not the domain(ping app.TeslaAndroid.com to get the ip.

Reboot the car also, it shouldn't be crashing frequently

**Reply by josbmw** [reply]
Ok, will try. Thnx.

**Reply by josbmw** [reply]
ABRP does not seem to work with the GPS receiver, no location. Google Maps and Waze work fine.
I'll have to wait until location is included in Tesla Android I guess...

**Reply by josbmw** [reply]
Any update on the new version with location info Mike?
